src/platform/backends/shared/base_snapshot.cpp

```diff
@@ -181,7 +181,7 @@ mp::BaseSnapshot::BaseSnapshot(const QJsonObject& json, VirtualMachine& vm)
           json["num_cores"].toInt(),                                                       // num_cores
           MemorySize{json["mem_size"].toString().toStdString()},                           // mem_size
           MemorySize{json["disk_space"].toString().toStdString()},                         // disk_space
-          MP_JSONUTILS.read_extra_interfaces(json),                                        // extra_interfaces
+          MP_JSONUTILS.read_extra_interfaces(json).value_or(desc.extra_interfaces),        // extra_interfaces
```

Comment:
There is only one step missing: you need to call `BaseSnapshot::persist` if the networks were empty. 

I think the most elegant way to achieve that is in the constructor being called here, i.e. the one in line 99. You would receive the optional there and call `value_or` in that other constructor. Then, in the body of the constructor, check if empty, assert captured and persist. Please let me know if that makes sense.

Reply:
Yes, that makes sense. Just to be clear, in your first statement: we need to call `persist()` if the networks were `nullopt`, not an empty vector, right?

Reply:
The only problem with doing that is that the constructor in line 99 (now 100) does not have information to infer the extra interfaces. Why not leaving that as it is in the current constructor, and calling `persist()` from the current constructor?

Adding a mew parameter to the constructor in line 99 would alter the rest of the code, because it is also called from another constructor which does not have the information needed.

Reply:
> Just to be clear, in your first statement: we need to call persist() if the networks were nullopt, not an empty vector, right?

Yep, that's right.

> The only problem with doing that is that the constructor in line 99 (now 100) does not have information to infer the extra interfaces. Why not leaving that as it is in the current constructor, and calling persist() from the current constructor?

Good point. OK, I guess you'll need to default-initialize the field and then add logic in the body of the constructor for this. I was trying to avoid that, but the alternatives aren't pretty either. I agree we shouldn't pass the new parameter to the other constructor and we should not add an another intermediate constructor for this either. So I think your proposal is a good compromise here.

Reply:
Good. What about what I implemented [here](https://github.com/canonical/multipass/pull/3439/commits/76b6b58243dcd675d8aedc888b320c642f98373b)?

Reply:
Oh OK. That's two json interrogations for the same field, but the alternative involves a 2-step "initialization" of the vector field, with possible growth, so your option is probably better :+1: